pkg/controller/jaeger/jaeger_controller.go

```diff
@@ -191,10 +191,15 @@ func (r *ReconcileJaeger) Reconcile(request reconcile.Request) (reconcile.Result
 	// update the status to "Ready"
 	if instance.Status.Phase != v1.JaegerPhaseRunning {
 		instance.Status.Phase = v1.JaegerPhaseRunning
-		if err := r.client.Status().Update(ctx, instance); err != nil {
-			logFields.WithError(err).Error("failed to store the running status into the current CustomResource")
-			return reconcile.Result{}, tracing.HandleError(err, span)
-		}
+	}
+
+	// set the status version to the updated instance version
+	instance.Status.Version = updated.Status.Version
```

Comment:
I don't think this is the right place to do this... We either do this during the instance creation, or after an upgrade.

Reply:
> This is missing a test. If you can reproduce the problem with a simple unit test, that's optimal, otherwise, an e2e test is required.

I'll add an e2e test, we have unit tests in place for the update process, even with those tests we didn't detect the issue before.